_posts/2022-07-25-changelog-175.markdown

```diff
@@ -0,0 +1,129 @@
+
+  {% include gif-img.html path="/assets/posts/changelog-175/json" w="700px" %}
+
+* [#9012] Introduce new approach of project model reloading and detection changes in configuration files
+  starting with 2022.2 major IDE release. Now the plugin detects changes in configuration files like
+  `Cargo.toml` even if they are not saved into disk yet, takes into account
+  [Cargo config](https://doc.rust-lang.org/cargo/reference/config.html),
+  [toolchain](https://rust-lang.github.io/rustup/overrides.html#the-toolchain-file) and
+  [build script](https://doc.rust-lang.org/cargo/reference/build-scripts.html) files and notifies you if
+  the project model is not up-to-date via floating button in the top-right corner of the editor.
+  You can select preferable way when the project model should be reloaded automatically in
+  `Preferences | Build, Execution, Deployment | Build Tools`. At the same time, Cargo settings were
+  moved to `Preferences | Build, Execution, Deployment | Build Tools | Cargo`
+
+  {% include gif-img.html path="/assets/posts/changelog-175/new-cargo-model-auto-import" w="700px" %}
+
+* [#8868] Highlight macro-machinery-specific syntax in `macro_rules` definitions.
+  New highlighting should help to distinguish a syntax specific to macros from
+  an expansion code (by [@daym])
+
+  <img src="/assets/posts/changelog-175/macro-highlighting.png" alt="Macro highlighting" width="700px"/>
+
+* [#8869] Enable by default some inspections for dependencies in `Cargo.toml`
+
+    * `Invalid crate version` warns you about a non-existing crate version
+
+      <img src="/assets/posts/changelog-175/invalid-crate-version.png" alt="Invalid crate version" width="700px"/>
+
+    * `New crate version available` notifies you about a crate that could be updated.
+
+      {% include gif-img.html path="/assets/posts/changelog-175/new-crate-version-available" w="700px" %}
+
+      We tried to make the highlighting not too intrusive, because using an older version of a crate
+      may be quite normal. If you still don't like the inspection, you can simply [disable it][disable-inspection].
+
+* Improve detection of some errors:
+
+    * [#9026] Improve detection of duplicate definitions.
+      Add support for [`E0254`](https://doc.rust-lang.org/error-index.html#E0254),
+      [`E0259`](https://doc.rust-lang.org/error-index.html#E0259) and
+      [`E0260`](https://doc.rust-lang.org/error-index.html#E0260) errors
+
+    * [#9020] Detect [E0430](https://doc.rust-lang.org/error-index.html#E0430)
+      `The self import appears more than once in the list` compiler error
+
+    * [#9000] Detect [E0742](https://doc.rust-lang.org/error-index.html#E0742)
+      `Visibility is restricted to a module which isn’t an ancestor of the current item` compiler error
+
+* [#9024] Support `View | Type Info` action for field shorthand syntax.
+  You can invoke `Type Info` action using <kbd>Ctrl</kbd> + <kbd>shift</kbd> + <kbd>P</kbd> shortcut
```

Comment:
Maybe just collapse action path and its shortcut like `Support `View | Type Info` action (<kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>P</kbd>)` as we usually do?